### PR TITLE
HSD8-500 Created UI for events importer to change URLs

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- UI for events importer
- removed site specific configs

# Needed By (Date)
- soon

# Urgency
- medium

# Steps to Test

1. Checkout this branch
1. Clear caches
1. Go do the events importer form at /admin/structure/migrate/events
1. Add an event url
1. Run importer
1. Verify new events content

- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: docroot/modules/humsci/hs_events/modules/hs_events_importer/src/Form/EventsImporterForm.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    parent::validateForm($form, $form_state);
+
+    $urls = array_filter(explode("\n", str_replace("\r", '', $form_state->getValue('urls'))));
```
Comment (sherakama) [inline docroot/modules/humsci/hs_events/modules/hs_events_importer/src/Form/EventsImporterForm.php]:
```suggestion
    $urls = array_filter(explode(PHP_EOL, str_replace(PHP_EOL, '', $form_state->getValue('urls'))));
```

PHP_EOL should be used for cross-platform compatibility.

Comment (sherakama) [reply docroot/modules/humsci/hs_events/modules/hs_events_importer/src/Form/EventsImporterForm.php]:
```suggestion
    $urls = array_filter(explode(PHP_EOL, $form_state->getValue('urls')));
```

File: docroot/modules/humsci/hs_events/modules/hs_events_importer/src/Form/EventsImporterForm.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+   * Validate that the user entered values are xml feeds from stanford events.
+   *
+   * @param string $url
+   *   Url to events.stanford.edu.
+   * @param array $form
+   *   Complete form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   Current form state.
+   */
+  protected function validateUrl($url, array &$form, FormStateInterface $form_state) {
```
Comment (sherakama) [inline docroot/modules/humsci/hs_events/modules/hs_events_importer/src/Form/EventsImporterForm.php:69]: Nice, thanks for this validation.

File: docroot/modules/humsci/hs_events/modules/hs_events_importer/src/Overrides/ConfigOverrides.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+   */
+  public function getCacheableMetadata($name) {
+    return new CacheableMetadata();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function createConfigObject($name, $collection = StorageInterface::DEFAULT_COLLECTION) {
+    return NULL;
```
Comment (sherakama) [inline docroot/modules/humsci/hs_events/modules/hs_events_importer/src/Overrides/ConfigOverrides.php:77]: Why is this function being nulled out?

File: config/archaeology/su_humsci_theme.settings.yml [hunk 4]
```diff
@@ -1,18 +0,0 @@
-features:
```
Comment (sherakama) [inline config/archaeology/su_humsci_theme.settings.yml:1]: This doesn't seem related, are you sure you want to remove this?

File: config/archaeology/config_split.config_split.archaeology.yml [hunk 5]
```diff
@@ -1,20 +0,0 @@
-uuid: 1e0bd1f9-ce83-4b7d-9e5c-d10011665b3d
```
Comment (sherakama) [inline config/archaeology/config_split.config_split.archaeology.yml:1]: Are you sure you want to remove this?

File: config/francestanford/config_split.config_split.francestanford.yml [hunk 6]
```diff
@@ -1,19 +0,0 @@
-uuid: 1e0bd1f9-ce83-4b7d-9e5c-d10011665b3d
```
Comment (sherakama) [inline config/francestanford/config_split.config_split.francestanford.yml:1]: This doesn't seem related, are you sure you want to remove this?

File: docroot/modules/humsci/hs_events/modules/hs_events_importer/src/Overrides/ConfigOverrides.php (hunk 3 above)
Comment (pookmish) [reply docroot/modules/humsci/hs_events/modules/hs_events_importer/src/Overrides/ConfigOverrides.php:77]: from the interface the docblock says:
> Creates a configuration object for use during install and synchronization.

we're not creating anything. only modifying a couple

File: config/archaeology/su_humsci_theme.settings.yml (hunk 4 above)
Comment (pookmish) [reply config/archaeology/su_humsci_theme.settings.yml:1]: Archaeology is now on the `archaeology` theme so these aren't even used now.

File: config/archaeology/config_split.config_split.archaeology.yml (hunk 5 above)
Comment (pookmish) [reply config/archaeology/config_split.config_split.archaeology.yml:1]: this PR will create a new config during update hook which replaces the need for the the site specific split

File: config/francestanford/config_split.config_split.francestanford.yml (hunk 6 above)
Comment (pookmish) [reply config/francestanford/config_split.config_split.francestanford.yml:1]: this PR will create a new config during update hook which replaces the need for the the site specific split

File: docroot/modules/humsci/hs_events/modules/hs_events_importer/src/Form/EventsImporterForm.php (hunk 1 above)
Comment (pookmish) [reply docroot/modules/humsci/hs_events/modules/hs_events_importer/src/Form/EventsImporterForm.php]: PHP_EOL is only the `\n` i found that while testing that there was also a `\r` at the end of lines: 
```
someurl\r\n
anotherurl
```

Comment (pookmish) [reply docroot/modules/humsci/hs_events/modules/hs_events_importer/src/Form/EventsImporterForm.php]: I replaced the \n, but i think we still need the \r replace.

Comment (sherakama) [reply docroot/modules/humsci/hs_events/modules/hs_events_importer/src/Form/EventsImporterForm.php]: TIL: https://stackoverflow.com/questions/3997336/explode-php-string-by-new-line

Yeah, looks like the browser submission takes lead on the line endings.

File: config/archaeology/config_split.config_split.archaeology.yml (hunk 5 above)
Comment (sherakama) [reply config/archaeology/config_split.config_split.archaeology.yml:1]: Thanks.

**Review by sherakama — APPROVED** [review]
GTG
